packages/ember-htmlbars/lib/keywords/component.js

```diff
@@ -22,6 +22,8 @@ export default {
       return;
     }
 
+    hash.isComponentHelper = true;
```

Comment:
You can add a flag above (in `setupState`) instead of modifying the hash.

Something like:

```javascript
setupState(lastState, env, scope, params, hash) {
    let state = {
      componentPath: env.hooks.getValue(params[0]),
      manager: lastState && lastState.manager,
      isComponentHelper: true
    };

    return state;
  }
```

Then access that in the `ComponentNodeManager.create` (it is called `renderNode` there) as `renderNode.isComponentHelper`.